api/v1/db/oracledb/reviews-dao.js

```diff
@@ -87,4 +88,75 @@ const getReviewById = async (id) => {
   }
 };
 
-module.exports = { getReviews, getReviewById };
+/**
+ * @summary Create review record
+ */
+const postReview = async (body) => {
+  const connection = await conn.getConnection();
+  try {
+    const { attributes } = body.data;
+    attributes.outId = { type: oracledb.NUMBER, dir: oracledb.BIND_OUT };
+    const sqlQuery = `
+      INSERT INTO REVIEWS
+      (REVIEWER, REVIEW_TEXT, SCORE, GAME_ID)
+      VALUES (:reviewer, :reviewText, :score, :gameId)
+      RETURNING ID INTO :outId
+    `;
+    const rawReviews = await connection.execute(sqlQuery, attributes, { autoCommit: true });
+
+    const result = await getReviewById(rawReviews.outBinds.outId[0]);
+    return result;
+  } finally {
+    connection.close();
+  }
+};
+
+/**
+ * @summary Delete review record
+ */
+const deleteReview = async (reviewId) => {
+  const sqlParams = { id: reviewId };
+  const sqlQuery = 'DELETE FROM REVIEWS WHERE ID = :id';
+
+  const connection = await conn.getConnection();
+  try {
+    const response = await connection.execute(sqlQuery, sqlParams, { autoCommit: true });
+    return response;
+  } finally {
+    connection.close();
+  }
+};
+
+/**
+ * @summary update review record
+ */
+const patchReview = async (reviewId, body) => {
+  const { attributes } = body.data;
+  // create iterator for the list of keys in attributes (parmeter names)
```

Comment:
This is a cool approach. It might be simpler to use the `join` method of an array to handle this logic though. For example, this produces the same query:

```
  const sqlSet = [`${attributes.reviewText ? 'REVIEW_TEXT = :reviewText' : ''}`,
                  `${attributes.score ? 'SCORE = :score' : ''}`,
                  `${attributes.score ? 'REVIEWER = :reviewer' : ''}`];
  const sqlQuery = `
    UPDATE REVIEWS
    SET ${sqlSet.join(',')}
    WHERE ID = :id
  `;
```

Reply:
@wilsonia wow, pretty clever!

Reply:
It almost works but the empty strings `''` are still considered an element so you get trailing commas when you join. Filtering those empty strings out makes it work though.
`filteredsqlSet = sqlSet.filter(element => element != '');`

Reply:
or maybe even simpler:

`filteredsqlSet = sqlSet.filter(element => element);`

base on the truth of this: https://stackoverflow.com/questions/8692982/in-javascript-is-an-empty-string-always-false-as-a-boolean